src/lib/kdf/kdf2/kdf2.cpp

```diff
@@ -20,41 +22,31 @@ std::unique_ptr<KDF> KDF2::new_object() const {
    return std::make_unique<KDF2>(m_hash->new_object());
 }
 
-void KDF2::kdf(uint8_t key[],
-               size_t key_len,
-               const uint8_t secret[],
-               size_t secret_len,
-               const uint8_t salt[],
-               size_t salt_len,
-               const uint8_t label[],
-               size_t label_len) const {
-   if(key_len == 0) {
+void KDF2::perform_kdf(std::span<uint8_t> key,
+                       std::span<const uint8_t> secret,
+                       std::span<const uint8_t> salt,
+                       std::span<const uint8_t> label) const {
+   if(key.empty()) {
       return;
    }
 
-   const size_t blocks_required = key_len / m_hash->output_length();
+   const size_t blocks_required = key.size() / m_hash->output_length();
+   BOTAN_ARG_CHECK(blocks_required < 0xFFFFFFFF, "KDF2 maximum output length exceeeded");
```

Comment:
`..FE`?

Reply:
Same discussion as above. 32-bit counter initialized at 1 this time. So 2^32-1 iterations should be safe (theoretically).